src/pybind/mgr/cephadm/agent.py

```diff
@@ -54,7 +55,18 @@ def __init__(self, mgr: "CephadmOrchestrator") -> None:
 
     def run(self) -> None:
         try:
-            self.ssl_certs.generate_root_cert()
+            try:
+                old_creds = self.mgr.get_store('cephadm_endpoint_credentials')
+                assert old_creds
+                old_creds_dict = json.loads(old_creds)
+                old_key = old_creds_dict['key']
+                old_cert = old_creds_dict['cert']
+                self.ssl_certs.load_root_credentials(old_cert, old_key)
+            except AssertionError:
+                self.ssl_certs.generate_root_cert()
+                for agent in self.mgr.cache.get_daemons_by_type('agent'):
+                    self.mgr._schedule_daemon_action(agent.name(), 'reconfig')
```

Comment:
we're scheduling actions for all agents here. At which point are we hitting the maximum value size of 64k in the k-v store?

Reply:
added some extra check when checking agent deps to verify if the root cert has changed or not to get around needing this